### DISCUSSION TITLE
How to expect a NextJS navigation event on the play function.

### DISCUSSION THREAD

**Discussion by conradogarciaberrotaran** [storybookjs/storybook#24330] [Help] [answered]
Hi everyone, I have a component that calls the replace method from next router.
I want to expect that this was called after a successful login, I'm not able to find how this is done:

My story looks like this:
```typescript
export default meta;
type Story = StoryObj<typeof LoginForm>;

export const SuccessfulLogin: Story = {
  parameters: {
    msw: {
      handlers: [successfulLoginHandler]
    },
  },
  play: async ({ canvasElement }) => {
    const canvas = within(canvasElement);
    const passwordInput = canvas.getByPlaceholderText('Password');
    const loginButton = canvas.getByText('Log in');

    expect(passwordInput).toBeInTheDocument();
    expect(loginButton).toBeInTheDocument();
    expect(loginButton).toBeDisabled();

    await userEvent.type(passwordInput, 'correctPassword');

    await waitFor(() => userEvent.click(loginButton));

    // TODO: Check router was called here

  },
};

```
My component is calling the 'nextNavigation.replace', I'm able to overwrite the replace method, but i'm still not able to expect that action.

Any help is appreciated.

**Comment by conradogarciaberrotaran** [comment]
bump?

**Comment by valentinpalkovic** [comment, marked as answer]
Unfortunately, this is currently not possible. But please look for a new package named `@storybook/test` in the next releases, which will allow you to test global spies, with which it should be possible to test navigation events in play functions.
